### DISCUSSION TITLE
Question about installing lazy vim

### DISCUSSION THREAD

**Discussion by JunkyardCat** [LazyVim/LazyVim#1445] [Q&A]
1. can lazyvim work on windows/termux(android)/linux(debian) completely? debians version of neovim seems to be a bit outdated
2. how do i uninstall lazyvim. (i had lazyvim installed and want to remove it)
3. can i ask where i can see common problem in installing lazyvim and how to remedy them?
4. can anyone guide me on the proper way to install a plugin? is it possible to manually install a plugin one by one up till it becomes neovim?

**Comment by dpetka2001** [comment]
1. Lazyvim is just a plugin for Neovim itself. So if Neovim can work in an environment, I'm guessing Lazyvim would as well (although I've only tried it on a Linux environment for my needs). 
2. How did you install Lazyvim? If you installed via the `starter` template, then you simply have to delete the directories that were created. If you installed via a package manager, then uninstall it from there.
3. I'm not sure I understand what you're asking here.
4. Usually, you install plugins via your package manager that you use (be it `lazy.nvim` or `packer` or the successor of `packer` which I don't remember its name).

**Reply by JunkyardCat** [reply]
@dpetka2001 
```
How did you install Lazyvim? If you installed via the starter template, then you simply have to delete the directories that were created. If you installed via a package manager, then uninstall it from there.
```
i instlled it doing `git clone https://github.com/LazyVim/starter $env:LOCALAPPDATA\nvim` and then `Remove-Item $env:LOCALAPPDATA\nvim\.git -Recurse -Force` then `nvim`

**Reply by dpetka2001** [reply]
Not familiar with Powershell, so just do the equivalent steps mentioned in the docs if you want to remove it. In linux these would be 
```
~/.config/nvim/
~/.local/share/nvim/
~/.local/state/nvim/
~/.cache/nvim/
```

**Reply by JunkyardCat** [reply]
does all plugin work in windows or do i need to do some sort of workaround?

**Reply by dpetka2001** [reply]
I don't use Windows, so I can't say for sure. You should read the plugins' docs to see what they support.

**Comment by hukadev** [comment]
Hello @dpetka2001 

Could you please help me to answer it?
I'm confusing between `lua/plugins/*.lua` and `lua/config/lazy.lua`.

For example: I want to add `mini.surround` plugin.
Then I have 2 options:

1. Import it in `lua/config/lazy.lua`
```
require("lazy").setup({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    { import = "mini.surround" },
    { import = "plugins" },
  },
})
```

2. Create a file `lua/plugins/mini.surround.lua`

The difference is if I want to config `mini.surround`, I can only choose Option 2, is it right?

**Reply by folke** [reply]
1. Mini.surround is already included in LazyVim, so there's no need to add it.
2. No need to change your lazy.lua file except when adding a LazyVim extra
3. You can't have a dot part of the filename of a Lua module. That's not a limitation of lazy. That's just how Lua works
4. Also, just check the docs on the lazyvim website. Specifically the section on plugins and examples. It's all In there

**Reply by dpetka2001** [reply]
@hukacode Folke is much more knowledgeable than me, so you should heed to his recommendations. Also, as he suggested, you should definitely read the docs (both Lazyvim and lazy.nvim) whenever you have some doubt, even if you'd already read them once.

**Reply by hukadev** [reply]
Thanks for your answer!
I really appreciate it!
